src/components/views/PasswordReset/NewPassword.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { message as alert } from 'antd';
+import 'antd/dist/antd.css';
+import { Link } from 'react-router-dom';
+import { applyReset } from '../../../redux/actions/passwordResetActions';
+
+const ResetPassword = ({
+  loading, success, message, resetPassword, match: { params },
+}) => {
+  const [formData, setFormData] = useState({
+    password: '',
+    confirmPassword: '',
+  });
+
+  const { password, confirmPassword } = formData;
+  const onChange = e => setFormData({ ...formData, [e.target.name]: e.target.value });
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const validatePassword = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})/;
+    const valid = validatePassword.test(password);
+    if (!valid) {
+      alert.error('Use a combination of uppercase, lowercase and numeric values');
+    } else if (password !== confirmPassword) {
+      alert.error('Passwords do not match!');
+    } else resetPassword(password, confirmPassword, params.token);
+  };
+
+  const error = !success && message;
+  return (
+    <div className="passwd-container">
+      <Link to="/">
+        <span className="brand-logo logo">
+          <img src="https://ci3.googleusercontent.com/proxy/gifkK5hB7Ze4QSQGIJOIc3y1s3KgYu0z2hw1-UiX4HQWd6tRnQwli7Ktka-q01P39puILmAoOoY_XVyUgA3Qz5vMIG644QCEmUBnWF5NAveFm1qUmj2b1rfJOjc45F-HEP9bSHau9SxoJ4xwdPnhOGBDJ7ksTLGm17O3zlNfqHs=s0-d-e1-ft#https://res.cloudinary.com/dsqyhgfws/image/upload/v1565811732/Screen_Shot_2019-08-14_at_8.41.34_PM_ssj6ej.png" alt="logo" />
+        </span>
+      </Link>
+      {
+        !success ? (
+          <form onSubmit={e => handleSubmit(e)}>
+            <h4>Create New Password</h4>
+            <div>
+              <p>Enter New Password</p>
+              <input
+                id="password-input"
+                name="password"
+                type="password"
+                onChange={e => onChange(e)}
+                minLength="6"
+              />
+            </div>
+
+            <div>
+              <p>Confirm Password</p>
+              <input
+                id="confirm-password"
+                name="confirmPassword"
+                type="password"
+                onChange={e => onChange(e)}
+                minLength="6"
+              />
+            </div>
+            <input
+              type="submit"
+              value="submit"
+              id="submit"
+              className="btn-block"
+            />
+          </form>
+        ) : (<h4>{message}</h4>)
+      }
+      { loading && <div className="loading"><h5>Loading...</h5></div> }
```

Comment:
I love that your `Loading...` works. Can you change the text to something more appropriate since your are not loading anything for the user per se? Also, try to disable the button as well

Reply:
Ok, I'll try that and see how it goes.